synapse/handlers/sso.py

```diff
@@ -52,3 +52,39 @@ def render_error(
             error=error, error_description=error_description
         )
         respond_with_html(request, 400, html)
+
+    async def get_sso_user_by_remote_user_id(
+        self, auth_provider_id: str, remote_user_id: str
+    ) -> Optional[str]:
+        """
+        Maps the user ID of a remote IdP to a mxid for a previously seen user.
+
+        If the user has not been seen yet, this will return None.
+
+        Args:
+            auth_provider_id: A unique identifier for this SSO provider, e.g.
+                "oidc" or "saml".
+            remote_user_id: The user ID according to the remote IdP. This might
+                be an e-mail address, a GUID, or some other form. It must be
+                unique and immutable.
+
+        Returns:
+            The mxid of a previously seen user.
+        """
+        # Check if we already have a mapping for this user.
+        logger.info(
+            "Looking for existing mapping for user %s:%s",
+            auth_provider_id,
+            remote_user_id,
+        )
```

Comment:
I question whether these need to be at `info` level.

Reply:
I suspect it is at info since a server admin might get a report "I can't login via SSO" or something and want to have *some* logs about what is happening.

(Note that this is also just copied, I didn't change the log level.)

I don't have a strong opinion on this though.

Reply:
> I suspect it is at info since a server admin might get a report "I can't login via SSO" or something and want to have some logs about what is happening.

That's a good point.

> (Note that this is also just copied, I didn't change the log level.)

Yeah, just questioning it as a drive-by. I have an idea for improvement, but I can drop these changes in a separate PR if you think they make sense:

Perhaps we could ditch/drop down the level of the "Looking" log line, and keep (and extend with more info) the "Found existing mapping" line for when users do have problems?

Reply:
That might work? I think it makes sense as a separate PR since that isn't really abstracting. 😄